### DISCUSSION TITLE
Change max upload size?

### DISCUSSION THREAD

**Discussion by Reziem** [jlesage/docker-sharry#2] [Q&A] [answered]
I see in the sharry documentation itself within the default config there is a.

```
 # Maximum size of a share.
      max-size = "1.5G"
```

How do I go about changing this in the docker container? I would love to increase this as my friends and I share large mod files for pc games.

**Comment by jlesage** [comment, marked as answer]
All Sharry configuration options can be set via environment variables. In your case, you would use `SHARRY_BACKEND_SHARE_MAX_SIZE=10G`. You can also edit the Sharry configuration file is you prefer. See https://github.com/jlesage/docker-sharry?tab=readme-ov-file#customizing-sharry-configuration for more details.

**Comment by NiyahVE** [comment]
Just found this, sorry to necropost but I would like to say that on unraid running last version of this docker, changing the file in /config/ doesn't do anything.

However, the env variable work, but it get a bit tiring to add env variable for everything I wish to change (default theme, logo, etc..)
Any idea ? :D

**Reply by jlesage** [reply]
> changing the file in /config/ doesn't do anything.

This should work. What is the problem exactly ? Your changes are lost ?

**Reply by NiyahVE** [reply]
My changes aren't lost, they just aren't taken into account.

For example, adding SHARRY_BACKEND_SHARE_MAX_SIZE=50G change the file limit to 50G, however if I add this:


```
    share {
      # When storing binary data use chunks of this size.
      chunk-size = "512K"

      # Maximum size of a share.
      max-size = "51G"

      # Maximum validity for uploads
      max-validity = 365 days

      # Allows additional database checks to be translated into some
      # meaningful message to the user.
      #
      # This config is used when inspecting database error messages.
      # If the error message from the database contains the defined
      # `native` part, then the server returns a 422 with the error
      # messages given here as `message`.
      #
      # See issue https://github.com/eikek/sharry/issues/255 – the
      # example is a virus check via a postgresql extension "snakeoil".
      database-domain-checks = {
        # Example: This message originates from postgres with an
        # enabled snakeoil extension. This extension allows to virus
        # check byte arrays. It must be setup such that the `bytea`
        # type of the filechunk table is changed to the type
        # `safe_bytea`:
        #
        # CREATE EXTENSION pg_snakeoil;
        # CREATE DOMAIN public.safe_bytea as bytea CHECK (not so_is_infected(value));
        # ALTER TABLE public.filechunk ALTER COLUMN chunkdata TYPE safe_bytea;
        snakeoil = {
          enabled = false
          native = "domain safe_bytea violates check constraint"
          message = "The uploaded file contains a virus!"
        }
      }
    }
```

Into the sharry.conf in volume in appdata/Sharry folder made by the docker, it's not applying the said changes.
    (With or without the env variable in, I added the variable after I found this discussion)

**Reply by jlesage** [reply]
Where exactly are you adding this in the config file ? It should be under `backend`.

**Reply by NiyahVE** [reply]
I was certain it was under backend but it wasn't, never do this kind of stuff without sleep !

Thanks you and sorry for wasting your time :)

**Reply by jlesage** [reply]
No problem!
